chat-sdk-core/src/main/java/co/chatsdk/core/dao/Message.java

```diff
@@ -254,6 +254,7 @@ public void setUserReadStatus (User user, ReadStatus status, DateTime date) {
         ReadReceiptUserLink link = linkForUser(user);
         if(link == null) {
             link = StorageManager.shared().createEntity(ReadReceiptUserLink.class);
+            link.setReadReceiptId(this.getId());
```

Comment:
Do you know if this step is necessary? Is it possible that the `readReceiptId` and `userId` need to be set to `Long` too?

Reply:
I was expecting greenDAO to resolve the foreign key automatically, but it did not. I also did not have much of a time to see why, so I just assigned it manually.

For the second question, yes I believe so.